tests/typed_ir/T620.links

```diff
@@ -0,0 +1,7 @@
+sig foo : () -> ()
```

Comment:
Looks like the only polymorphism here are the effect variables hidden by using -> and ~>. Maybe we should make them explicit here, as their handling is what this test actually checks.

Reply:
I don't mind adding more tests, if there are any cases to be tested, but I definitely want to include the program that was originally reported in the bug report.

Overall, I'm not too worried about this test. Once we fix all the IR typing error we can enable IR checking in Travis/Azure and that will provide an extensive testing of IR correctness of programs.